src/mina-signer/tests/rosetta.test.ts

```diff
@@ -23,6 +25,25 @@ describe('Rosetta', () => {
     client = new Client({ network: 'mainnet' });
   });
 
+  it('generates a valid signature from a hex string', () => {
+    let { r, s } = signatureFromHex(rosettaTnxMockSignature);
+    expect(r).not.toBeNaN();
+    expect(r).not.toBeFalsy();
+    expect(s).not.toBeNaN();
+    expect(s).not.toBeFalsy();
+
+    // TODO we could go further and use validate method on the return?
+    // or leave as is with the assumption that the above hex is known to be valid
```

Comment:
yes - ideally, we should verify the signature against the payment, using `signedRosettaTransactionToSignedCommand` and `verifyPaymentSignature` (or however it's called). that's a big hole we left in the tests when we created this

Reply:
even if the hex above is valid -- only by verifying the signature can we really be sure that we parse it correctly

Reply:
i'm going to start down this path now. getting verification into the test. perhaps along the way we can reveal why that test fails?

question: do you know if this hex, the "signature" in the mock - was it generated with the ocaml signer? 

re: we know that the ocaml signer produces output (hex string) that is compatible with the rosetta `construction/submit` enpoint. we know this from the .py scripts written by Ivan that are in the repo as examples 

thus: if we have a mock signature in a known valid format we can proceed in our tests with confidence that when i get to actually trying to POST to /construction/submit we will (likely) succeed. that's my hope anyway

Reply:
note: i did find that the test-vectors used by the unit tests in sign-legacy function correctly..

Reply:
I assume I copied the mock from OCaml test vectors but can't remember